python/pyiceberg/exceptions.py

```diff
@@ -90,3 +90,19 @@ class SignError(Exception):
 
 class ResolveError(Exception):
     pass
+
+
+class DynamoDbError(Exception):
+    pass
+
+
+class ItemNotFound(DynamoDbError):
```

Comment:
What do you think of replacing this one with the build in ValueError? It is not being used a lot, and we want to be mindful of introducing new exceptions.

Reply:
Done.